File: python/IoT platform/Mobius_APIs_Python/onem2m_post_cin_img.py
# Created by J. Yun, SCH Univ., yun@sch.ac.kr

import requests
import base64

url = 'http://203.253.128.161:7579/Mobius/sch19999999/img'
headers =	{
				'Accept':'application/json',
				'X-M2M-RI':'12345',
				'X-M2M-Origin':'XXXXXXXXXXX', # change to your aei
				'Content-Type':'application/vnd.onem2m-res+json; ty=4'
			}

with open("oneM2M_logo.png", "rb") as img:
    enc_str = base64.b64encode(img.read()).decode('utf-8')
# print(enc_str)

# print(enc_str)
data =	{
			"m2m:cin": {
				"con": enc_str
			}
		}

r = requests.post(url, headers=headers, json=data)

try:
	r.raise_for_status()
	jr = r.json()
	
	print(jr)
	print(jr['m2m:cin']['con'])
except Exception as exc:
	print('There was a problem: %s' % (exc))

File: nodejs/onem2m_post_cin_img.js
// Created by J. Yun, SCH Univ., yun@sch.ac.kr
// Need to install node-fetch
// $ npm i node-fetch --save

const fetch = require("node-fetch");

var myHeaders = new fetch.Headers();
myHeaders.append("Accept", "application/json");
myHeaders.append("X-M2M-RI", "12345");
myHeaders.append("X-M2M-Origin", "XXXXXXXXXXX"); // change to your aei
myHeaders.append("Content-Type", "application/vnd.onem2m-res+json; ty=4");

var fs = require("fs");
var enc_str = fs.readFileSync("./oneM2M_logo.png", { encoding: "base64" });
var raw = '{"m2m:cin": {"con": "' + enc_str + '1"}}';

var requestOptions = {
  method: "POST",
  headers: myHeaders,
  body: raw,
  redirect: "follow",
};

fetch("http://203.253.128.161:7579/Mobius/sch19999999/img", requestOptions)
  .then(function (response) {
    if (response.status !== 201) {
      console.log("There was a problem. Status Code: " + response.status);
      return;
    }

    response.json().then(function (data) {
      console.log(data);
      console.log(data["m2m:cin"]["con"]);
    });
  })
  .catch(function (err) {
    console.log("Fetch Error :-S", err);
  });
